File: lib/core_services/eiga/mixin/eiga_follow_mixin.dart
import 'package:hoyomi/core_services/eiga/export.dart';
import 'package:hoyomi/core_services/mixin/auth_mixin.dart';

mixin EigaFollowMixin on AuthMixin {
  Future<Paginate<EigaFollow>> getFollows({required int page});

  Future<int> getFollowsCount(String eigaId, MetaEiga metaEiga) {
    throw UnimplementedError();
  }

  Future<bool> isFollow(String eigaId);
  Future<void> setFollow(EigaContextWithEpisodes context, bool value);
}

File: lib/core_services/eiga/mixin/eiga_follow_mixin.ts
import type { Paginate } from "../../interfaces/paginate"
import { AuthMixin } from "../../mixin/auth_mixin"
import type { EigaFollow } from "../interfaces/eiga_follow"
import type { EigaContextWithEpisodes } from "../interfaces/eiga_context_with_episodes"

export abstract class EigaFollowMixin extends AuthMixin {
  abstract getFollows(page: number): Promise<Paginate<EigaFollow>>

  abstract getFollowsCount(context: EigaContextWithEpisodes): Promise<number>

  abstract isFollow(eigaId: string): Promise<boolean>

  abstract setFollow(
    context: EigaContextWithEpisodes,
    value: boolean
  ): Promise<void>
}
